src/BumpChart.js

```diff
@@ -0,0 +1,69 @@
+import {extent} from "d3-array";
+import {constant} from "d3plus-common";
+
+import {default as Plot} from "./Plot";
+
+/**
+    @class BumpChart
+    @extends Plot
+    @desc Creates a bump chart based on an array of data.
+    @example <caption>the equivalent of calling:</caption>
+new d3plus.Plot()
+  .discrete("x")
+  .shape("Line")
+  .x("x")
+  .y2("y")
+  .yConfig({
+    tickFormat: val => {
+      const data = this._filteredData;
+      const xDomain = extent(data.map(d => d.x));
```

Comment:
Using the `extent` of the x axis values won't always work, as it's a discrete axis and discrete axis values don't necessarily have to be numbers! Let's have `Viz.js` store the axis domains (which it already calculates) in `this` so that we can just access the domain directly.